cpp/include/cuvs/cluster/agglomerative.hpp

```diff
@@ -144,6 +145,8 @@ struct mutual_reachability_params {
   /** weight applied when internal distance is chosen for mutual reachability (value of 1.0 disables
    * the weighting) */
   float alpha = 1.0;
+  cuvs::neighbors::all_neighbors::all_neighbors_params all_neighbors_p{
```

Comment:
Can we stick to the suffix `_params`? `all_neighbors_p` can be slightly confusing. Also add docs for this struct here!